eta/core/numutils.py

```diff
@@ -71,12 +71,21 @@ class GrowableArray(object):
     '''A class for building a numpy array from streaming data.'''
 
     def __init__(self, rowlen):
+        '''Creates a GrowableArray.
+
+        Args:
+            rowlen: the desired length of each row
+        '''
         self.data = []
         self.rowlen = rowlen
 
     def update(self, row):
         '''Add row to array.'''
-        assert len(row) == self.rowlen, "Expected row length %d" % self.rowlen
+        if len(row) != self.rowlen:
```

Comment:
Unclear to me what the difference is in functionality or error-checking.

Reply:
just following the developing convention to raise custom exceptions that categorize our errors